thrift/server.hh

```diff
@@ -35,6 +35,12 @@ class thrift_server;
 class thrift_stats;
 class database;
 
+#ifdef THRIFT_USES_BOOST
+#define shared_ptr_ns boost
+#else
+#define shared_ptr_ns std
+#endif
+
```

Comment:
Again no #defines. You can do this just once, and use thrift_std, instead of defining multiple names.